examples/ssr-data-router/src/entry.client.tsx

```diff
@@ -6,9 +6,31 @@ import { routes } from "./App";
 
 let router = createBrowserRouter(routes);
 
-ReactDOM.hydrateRoot(
-  document.getElementById("app"),
-  <React.StrictMode>
-    <RouterProvider router={router} fallbackElement={null} />
-  </React.StrictMode>
-);
+// If you're using lazy route modules and you haven't yet preloaded them onto
+// routes, then you'll need to wait for the router to be initialized before
+// hydrating, since it will have initial data to hydrate but it won't yet have
+// any router elements to render.
+//
+// This shouldn't be needed in most SSR stacks as you should know what routes
+// are initially rendered and be able to SSR the appropriate <Script> tags for
+// those modules such that they're readily available on your client-side route
+// definitions
+if (!router.state.initialized) {
+  let unsub = router.subscribe((state) => {
+    if (state.initialized) {
+      unsub();
+      hydrate();
+    }
+  });
+} else {
+  hydrate();
+}
```

Comment:
Minor changes needed to leverage this during SSR hydration _only if your SSR stack is unable to preload these modules_.  Really, any SSR stack should create a client side routing tree where the initial routes are prefilled and don't have `lazy` on them.

Reply:
Could potentially wrap this up in an `onInitialize` callback passed to `createBrowserRouter` and into `createRouter`